src/details/ArborX_DetailsHappyTreeFriends.hpp

```diff
@@ -75,9 +75,26 @@ struct HappyTreeFriends
     return bvh._internal_and_leaf_nodes(i).left_child;
   }
 
+  template <class BVH>
+  static KOKKOS_FUNCTION auto getRightChild(BVH const &bvh, int i)
+  {
+#ifndef KOKKOS_ENABLE_CXX17
+    return getRightChildImpl(bvh, i);
+#else
+    static_assert(has_node_with_two_children<BVH>::value ||
+                  has_node_with_left_child_and_rope<BVH>::value);
+    assert(!isLeaf(bvh, i));
+    if constexpr(has_node_with_left_child_and_rope<BVH>::value)
+      return bvh._internal_and_leaf_nodes(getLeftChild(bvh, i)).rope;
+    else
+      return bvh._internal_and_leaf_nodes(i).right_child;
```

Comment:
Wonder if clang-tidy would complain to make you do this:
```suggestion
    if constexpr(has_node_with_left_child_and_rope<BVH>::value)
      return bvh._internal_and_leaf_nodes(getLeftChild(bvh, i)).rope;
    return bvh._internal_and_leaf_nodes(i).right_child;
```

Reply:
It would not compile
https://godbolt.org/z/xhETqG4Pn

Reply:
I am dumb. `constexpr` ifs are different from regular ones.